### PR TITLE
Add Date to Terms and Conditions

### PR DESCRIPTION
**Pull request by Tirokk**
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-09-09T12:09:17Z" title="Monday, September 9th 2019, 2:09:17 pm +02:00">Sep 9, 2019</time>_
_Closed <time datetime="2019-09-12T16:10:11Z" title="Thursday, September 12th 2019, 6:10:11 pm +02:00">Sep 12, 2019</time>_
---

## 🍰 Pullrequest
When confirming the current VERSION of the terms and conditions, the DATE is also saved (termsAndConditionsAgreedAt). 

### Issues
- fixes https://github.com/Human-Connection/Human-Connection/pull/1556

### CODE REVIEW

File: backend/src/middleware/slugifyMiddleware.spec.js [hunk 1]
```diff
@@ -1,81 +1,62 @@
+import { GraphQLClient } from 'graphql-request'
 import Factory from '../seed/factories'
-import { gql } from '../jest/helpers'
-import { neode as getNeode, getDriver } from '../bootstrap/neo4j'
-import createServer from '../server'
-import { createTestClient } from 'apollo-server-testing'
```
Comment (Tirokk) [inline backend/src/middleware/slugifyMiddleware.spec.js:5]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-09T13:05:41Z" title="Monday, September 9th 2019, 3:05:41 pm +02:00">Sep 9, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -89,8 +89,8 @@ describe('slugify', () => {
   })
 
   describe('SignupVerification', () => {
-    const mutation = `mutation($password: String!, $email: String!, $name: String!, $slug: String, $nonce: String!, $termsAndConditionsAgreedVersion: String!) {
-      SignupVerification(email: $email, password: $password, name: $name, slug: $slug, nonce: $nonce, termsAndConditionsAgreedVersion: $termsAndConditionsAgreedVersion) { 
+    const mutation = `mutation($password: String!, $email: String!, $name: String!, $slug: String, $nonce: String!, $termsAndConditionsAgreedVersion: String!, termsAndConditionsAgreedAt: String) {
```


No, the `termsAndConditionsAgreedAt` should *not* be supplied by the user. We don't care what the user tells us, what date it is. We **know** the date *on the server*. So, in the `SignupVerification` resolver, we set the `termsAndConditionsAgreedAt` manually.

**Comment by Tirokk** [comment]
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-09-10T06:26:19Z" title="Tuesday, September 10th 2019, 8:26:19 am +02:00">Sep 10, 2019</time>_

---


i have it now so ... that the date is saved to the version. however at no place can be displayed. even :4000 cannot display it .. but :7687 there i see it at the user ...

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-11T15:01:45Z" title="Wednesday, September 11th 2019, 5:01:45 pm +02:00">Sep 11, 2019</time>_

---


> maybe for consistency we keep this in `backend/src/middleware/dateTimeMiddleware.js`
> although, we are having a bug with `createdAt` so what do ya'll think?

never mind @ogerly, I'll refactor to use `createdAt` and `updatedAt` in the resolvers themselves :smile:

**Review by Mogge — CHANGES_REQUESTED** [review]
@ogerly I think you got me wrong: The server should know the current date already. So we don't have to expose the `termsAndConditionsAgreedAt` as input param to the client. The attribute is certainly read-only, maybe not even that.

**Review by Mogge — CHANGES_REQUESTED** [review]
maybe for consistency we keep this in `backend/src/middleware/dateTimeMiddleware.js`
although, we are having a bug with `createdAt` so what do ya'll think?

**Pull request closed without merging**